src/Identity.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         2.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+declare(strict_types=1);
+
+/**
+ * An identity decorator implementing both Authorization\IdentityInterface and
+ * Authentication\IdentityInterface.
+ */
+namespace Authorization;
+
+use Authentication\IdentityInterface as AuthenIdentityInterface;
+
+class Identity extends IdentityDecorator implements AuthenIdentityInterface
+{
+    /**
+     * Get the primary key/id field for the identity.
+     *
+     * @return string|int|null
+     */
+    public function getIdentifier()
+    {
+        if ($this->identity instanceof AuthenIdentityInterface) {
+            return $this->identity->getIdentifier();
+        }
+
+        return $this->identity['id'];
```

Comment:
This should probably be configurable.

Reply:
Generally this line will never be executed as the class is only meant to be used when `authentication` plugin is used and hence the `$identity` property should be `Authentication\IdentityInterface` instance. I probably wrote this this way initially to just make running the test suite easier. 

I can change the method to simply `return $this->identity->getIdentifier();` and tests run fine now. Users can always specify their own decorator class for the `AuthorizationMiddleware` if this doesn't work for them.

Keeping the method as it is and allowing `id` to be configurable will requiring changing how `identityDecorator` config can be set. Currently it can just be a string. I don't really feel like going about making changes for it.

Reply:
I understand. So maybe throw an exception then?

Reply:
Sounds good, will do.

Reply:
I took a slightly different approach and modified the constructor so that it only takes `Authentication\IdentityInterface` as argument. That avoid any additional checks or exceptions in methods.